runtime/amd64.S

```diff
@@ -1132,10 +1128,9 @@ LBL(112):
            original performer stack. For that, we perform the necessary calls
            to __tsan_func_entry via caml_tsan_entry_on_resume. */
         TSAN_SAVE_CALLER_REGS
-        movq    Stack_sp(%r10), %r11
```

Comment:
Is the removal of that assignment to `%r11` intended?

Reply:
Yes, I think this should be correct: we execute a `SWITCH_OCAML_STACKS` above which restores `rsp` for the new fiber:
https://github.com/ocaml/ocaml/blob/72d9b1e03c175365d9eaf98a75493001bc38015b/runtime/amd64.S#L309
and right after that, we use `rsp` directly instead of  `r11` to access the stack return address and argument.

Reply:
I agree that this looks correct, but then if we want to rely on this we may also use the fact that `%r10` was preserved by SWITCH_OCAML_STACKS, and use it instead of `Caml_state(current_stack)` below? Why did you change the use of `%r10` in the current trunk, and I missing something and it became incorrect?

(In any case you could include a small comment here to point out where `%rsp` (and `%r10` if relevant) come from.)

Reply:
I find it redundant to comment the value of rsp when rsp plays the usual role of stack pointer. The fact that we are now on the performer fiber due to `SWITCH_OCAML_STACKS` is rather clear by the name of the macro.

I agree with you that we could reuse r10, however (and then a small comment to remind of its value makes sense to me).